### PR TITLE
Sync Volar 1.9

### PR DESCRIPTION
**Pull request by johnsoncodehk**
Hi, I tried to make the required changes for the TS Plugin integration updated to Volar 1.9, can you see if it matches your original needs? Thanks.

### CODE REVIEW

**Comment by mizdra** [comment]
Thank you! I was going to update Volar.js too, but didn't have time to do it. So sending me the PR helps me a lot.

 I will check the PR now.

File: src/index.ts [hunk 1]
```diff
@@ -15,38 +19,19 @@ const init: ts.server.PluginModuleFactory = (modules) => {
...
-            property === 'getTypeDefinitionAtPosition' ||
-            property === 'getImplementationAtPosition' ||
-            property === 'getReferencesAtPosition' ||
-            property === 'findReferences'
-          ) {
-            return assetLS[property];
-          }
-          return target[property];
-        },
-      });
```
Comment (mizdra) [inline src/index.ts:43]: I am glad to remove many of the codes :)

**Comment by mizdra** [comment]
I found a problem with newly added assets not being suggested due to forgetting to implement `getScriptFileNames`. I fixed the implementation.: https://github.com/mizdra/typescript-plugin-asset/pull/7/commits/db4818287da64398320de2b50b8744823f6897e2

Other than that, there seems to be no problem. It matches my needs well enough. Thanks! @johnsoncodehk

**Comment by mizdra** [comment]
I will merge this PR if it is ok with you.

**Comment by johnsoncodehk** [comment]
Hi, what problems can it cause if we don't implement `getScriptFileNames`? Can you explain the steps to reproduce the problem?
I'm asking this question because `getExternalFiles` is already implemented, so `getScriptFileNames` is theoretically not needed.

**Comment by mizdra** [comment]
If the patch https://github.com/mizdra/typescript-plugin-asset/commit/db4818287da64398320de2b50b8744823f6897e2 is not included, the added assets will not be suggested.

How to reproduce:
```console
$ git checkout 6c19f97
$ npm run build

$ # Open example project with VS Code
$ cd examples/nextjs
$ code .
```

https://github.com/mizdra/typescript-plugin-asset/assets/9639995/91e49908-6fd5-4c6c-868b-9925fbe88660

If the patch is included, the added assets are also suggested.


https://github.com/mizdra/typescript-plugin-asset/assets/9639995/46ff9146-0f63-4bfa-9892-c47e35cba45e

**Comment by johnsoncodehk** [comment]
Thanks! I'll investigate how to fix it from the framework, before that I think we can merge this PR, as the change is probably just an implementation improvement rather than a bug fix.

**Comment by mizdra** [comment]
It sounds good that Volar.js would take care of it :+1:. 

I'll merge this PR for now. Thanks for the contribution!

**Pull request merged**